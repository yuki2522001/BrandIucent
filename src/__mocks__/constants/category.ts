import { Categories } from "@common-types/category";
import url from "@assets/images/icons/beverages.png";

const CATEGORY: Categories = {
  id: "1234",
  name: "Beverages",
  images: {
    src: url,
    alt: "This is icon beverages",
  },
};

const CATEGORY_MOCKING_LIST: Categories[] = [
  {
    id: "1234",
    name: "Beverages",
    images: {
      src: "@assets/images/icons/beverages.png",
      alt: "This is icon beverages",
    },
  },
  {
    id: "12345",
    name: "Dairy",
    images: {
      src: "@assets/images/icons/dairy.png",
      alt: "This is icon dairy",
    },
  },
  {
    id: "12345",
    name: "Meat",
    images: {
      src: "@assets/images/icons/meat.png",
      alt: "This is icon meat",
    },
  },
  {
    id: "12345",
    name: "Spices & Herbs",
    images: {
      src: "@assets/images/icons/spices-herbs.png",
      alt: "This is icon spices-herbs",
    },
  },
  {
    id: "12345",
    name: "Stationery",
    images: {
      src: "@assets/images/icons/stationery.png",
      alt: "This is icon stationery",
    },
  },
  {
    id: "12345",
    name: "Costmetic",
    images: {
      src: "@assets/images/icons/costmetic.png",
      alt: "This is icon costmetic",
    },
  },
];

export { CATEGORY, CATEGORY_MOCKING_LIST };
