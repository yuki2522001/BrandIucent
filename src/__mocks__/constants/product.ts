import { Product } from "@common-types/product";
import iceCreamSundae from "@assets/images/ice-cream-sundae.png";
import beefSteak from "@assets/images/beef-steak.png";

const PRODUCT_MOCKING: Product = {
  productId: "16466545464646547",
  categoryId: "1651454657522",
  productName: "Ice Cream Sundae",
  productDescription:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam molestie ultrices donec pulvinar quis est bibendum suspendisse viverra.",
  productQuantity: 5,
  originalPrice: {
    value: 20,
    currency: "$",
  },
  discountPrice: {
    value: 20,
    currency: "$",
  },
  offerQuantity: "3kg",
  offer: false,
  bestSelling: true,
  popular: false,
  images: {
    src: iceCreamSundae,
    alt: "This is ice-cream-sundae images",
  },
};
const PRODUCT_MOCKING_LIST: Product[] = [
  {
    productId: "16466545464646547",
    categoryId: "1651454657522",
    productName: "Ice Cream Sundae",
    productDescription:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam molestie ultrices donec pulvinar quis est bibendum suspendisse viverra.",
    productQuantity: 5,
    originalPrice: {
      value: 20,
      currency: "$",
    },
    discountPrice: {
      value: 20,
      currency: "$",
    },
    offerQuantity: "3kg",
    offer: false,
    bestSelling: true,
    popular: false,
    images: {
      src: iceCreamSundae,
      alt: "This is ice-cream-sundae images",
    },
  }
];

export { PRODUCT_MOCKING, PRODUCT_MOCKING_LIST };
