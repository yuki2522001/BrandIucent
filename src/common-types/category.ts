export interface Categories {
  id: string;
  name: string;
  images: {
    src: string;
    alt: string;
  };
}