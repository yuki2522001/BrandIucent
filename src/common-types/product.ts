export interface Product {
  productId: string;
  categoryId: string;
  productName: string;
  productDescription: string ;
  productQuantity: number;
  originalPrice: {
    value: number;
    currency: string;
  };
  discountPrice: {
    value: number;
    currency: string;
  };
  offerQuantity: string;
  offer: boolean;
  bestSelling: boolean;
  popular: boolean;
  images: {
    src: string;
    alt: string;
  }
}

export interface ProductProps {
  product?: Product;
}