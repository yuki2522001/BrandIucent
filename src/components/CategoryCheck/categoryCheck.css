input[type="checkbox"] {
  -webkit-appearance: none;
  appearance: none;
  background-color: var(--light);
  margin: 0;
  appearance: none;
  font: inherit;
  width: 20px;
  height: 20px;
  border: 3px solid var(--dark);
  border-radius: 2px;
  display: grid;
  place-content: center;
  transform: translateY(-9px);
}

.checkbox__value {
  display: flex;
  align-items: flex-end;
}

.checkbox__value > input {
  margin-right: 25px;
}

input[type="checkbox"]::before {
  content: "";
  width: 0.65em;
  height: 0.65em;
  transform: scale(0);
  box-shadow: inset 1em 1em var(--dark);
  clip-path: polygon(14% 44%, 0 65%, 50% 100%, 100% 16%, 80% 0%, 43% 62%);
}

input[type="checkbox"]:checked::before {
  transform: scale(1);
}