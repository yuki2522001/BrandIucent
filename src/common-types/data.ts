import { Categories } from "./category";
import { Product } from "./product";

export interface DataContextProps {
  products: Product[];
  setProducts: Function;
  searchValue: [] | undefined;
  setSearchValue: Function;
  categories: Categories[];
  setCategories: Function;
}