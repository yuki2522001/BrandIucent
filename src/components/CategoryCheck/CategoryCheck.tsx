import React, { memo, MouseEvent } from "react";
import "./categoryCheck.css";

interface CategoryCheckProps {
  text: string;
  categoryId: string;
  value: boolean;
  onToggleCategory: Function;
  checked: boolean;
}

const CategoryCheck: React.FC<CategoryCheckProps> = memo(
  ({ text, onToggleCategory, categoryId, checked }) => {
    const handleCheck = (
      event: MouseEvent<HTMLInputElement> & { target: HTMLInputElement }
    ) => {
      const value = event.target.checked;
      onToggleCategory(categoryId, value);
    };

    return (
      <label data-testid="category-check" className="checkbox__value">
        <input
          type="checkbox"
          name="checkbox"
          multiple
          defaultChecked={checked}
          data-index={categoryId}
          onClick={handleCheck}
        />
        {text}
      </label>
    );
  }
);

export default CategoryCheck;
