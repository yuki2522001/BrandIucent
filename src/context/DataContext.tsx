import { createContext, useMemo, useState } from "react";
import { Categories } from "@common-types/category";
import { DataContextProps } from "@common-types/data";
import { Product } from "@common-types/product";

const DataContext = createContext<DataContextProps>({} as DataContextProps);
const DataProvider: React.FC<{ children: JSX.Element[] | JSX.Element }> = ({
  children,
}) => {
  const [searchValue, setSearchValue] = useState<[]>([]);
  const [categories, setCategories] = useState<Categories[]>([]);
  const [products, setProducts] = useState<Product[]>([]);

  const value = {
    products,
    setProducts,
    searchValue,
    setSearchValue,
    categories,
    setCategories,
  };

  return <DataContext.Provider value={value}>{children}</DataContext.Provider>;
};

export { DataProvider, DataContext };
