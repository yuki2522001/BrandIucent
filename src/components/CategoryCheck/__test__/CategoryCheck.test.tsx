import { render } from "@testing-library/react";
import { createMemoryHistory } from "history";
import { Router } from "react-router-dom";
import { DataContextProps } from "@common-types/data";
import { DataContext } from "@context/DataContext";
import { CATEGORY, CATEGORY_MOCKING_LIST } from "@__mocks__/constants/category";
import { PRODUCT_MOCKING_LIST } from "@__mocks__/constants/product";
import "@testing-library/jest-dom";
import CategoryCheck from "../CategoryCheck";

const contextProductMock: DataContextProps = {
  searchValue: [],
  setSearchValue: jest.fn(),
  categories: CATEGORY_MOCKING_LIST,
  setCategories: jest.fn(),
  products: PRODUCT_MOCKING_LIST,
  setProducts: jest.fn(),
};

describe("Categories component", () => {
  test("should render categories component", () => {
    const history = createMemoryHistory();
    const { getByTestId } = render(
      <DataContext.Provider value={contextProductMock}>
        <Router location={history.location} navigator={history}>
          <CategoryCheck
            text="Dairy"
            categoryId={""}
            value={false}
            onToggleCategory={() => {}}
            checked={false}
          />
        </Router>
      </DataContext.Provider>
    );
    expect(getByTestId("category-check")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const history = createMemoryHistory();
    const { asFragment } = render(
      <DataContext.Provider value={contextProductMock}>
        <Router location={history.location} navigator={history}>
          <CategoryCheck
            text="Dairy"
            categoryId={""}
            value={false}
            onToggleCategory={() => {}}
            checked={false}
          />
        </Router>
      </DataContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
