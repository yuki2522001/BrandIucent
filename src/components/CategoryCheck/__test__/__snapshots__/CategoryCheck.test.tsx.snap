// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories component matches snapshot 1`] = `
<DocumentFragment>
  <label
    class="checkbox__value"
    data-testid="category-check"
  >
    <input
      data-index=""
      multiple=""
      name="checkbox"
      type="checkbox"
    />
    Dairy
  </label>
</DocumentFragment>
`;
